### PR TITLE
feat: Add factory methods for AddEventRequest, Event and Reading DTOs

### PR DESCRIPTION
**Pull request by lenny-intel**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/go-mod-core-contracts/blob/master/.github/Contributing.md.

## What is the current behavior?
AddEventRequest, Event and Reading DTOs do not have factory methods

## What is the new behavior?
AddEventRequest, Event and Reading DTOs now have factory methods

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## New Imports

- [ ] Yes
- [x] No

## Specific Instructions
Are there any specific instructions or things that should be known prior to reviewing?

## Other information

### CODE REVIEW

File: v2/dtos/event.go [hunk 1]
```diff
@@ -28,6 +31,17 @@ type Event struct {
 	Tags               map[string]string `json:"tags,omitempty" xml:"-"` // Have to ignore since map not supported for XML
 }
 
+// NewEvent creates and returns an initialized Event with no Readings
+func NewEvent(profileName string, deviceName string) Event {
+	return Event{
+		Versionable: common.NewVersionable(),
+		Id:          uuid.New().String(),
```
Comment (cloudxxx8) [inline v2/dtos/event.go:38]: we can start using the equivalent new function `uuid.NewString()`

File: v2/dtos/common/base.go [hunk 2]
```diff
@@ -14,6 +18,10 @@ type BaseRequest struct {
 	RequestId string `json:"requestId" validate:"len=0|uuid"`
 }
 
+func NewBaseRequest() BaseRequest {
+	return BaseRequest{RequestId: uuid.New().String()}
```
Comment (cloudxxx8) [inline v2/dtos/common/base.go:22]: we need ApiVersion more than RequestId here

Comment (lenny-intel) [reply v2/dtos/common/base.go:22]: BaseRequest currently doesn't contain ApiVersion, the sub-DTOs (Event & Reading) do contain the ApiVersion. I will add ApiVersion to BaseRequest so it is consistent with BaseResponse and sub-DTOs.

**Review by cloudxxx8 — DISMISSED** [review]
LGTM

**Review by cloudxxx8 — APPROVED** [review]
LGTM

**Pull request merged**